src/features/users/users.ts

```diff
@@ -0,0 +1,34 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+
+export const fetchUsers = createAsyncThunk('users/fetch', () => getUsers());
+
+type InitialState = {
+  users: User[];
+  selectedUser: User | null;
+};
+const initialState: InitialState = {
+  users: [],
+  selectedUser: null,
+};
```

Comment:
Consider adding `loaded` and `hasError` properties to the initial state to handle loading states and errors consistently, similar to other slices. This will help manage UI states effectively.